src/address.rs

```diff
@@ -207,53 +133,21 @@ impl<A, Rc: RefCounter> Address<A, Rc> {
         }
     }
 
-    /// Send a [`Message`](../trait.Message.html) to the actor without waiting for a response.
-    /// If the actor's mailbox is full, it will block. If this returns `Err(Disconnected)`, then the
-    /// actor is stopped and not accepting messages. If this returns `Ok(())`, the will be delivered,
-    /// but may not be handled in the event that the actor stops itself (by calling
-    /// [`Context::stop`](../struct.Context.html#method.stop)) before it was handled.
-    pub fn do_send<M>(&self, message: M) -> Result<(), Disconnected>
-    where
-        M: Send + 'static,
-        A: Handler<M>,
-    {
-        if self.is_connected() {
-            // To read more about what an envelope is and why we use them, look under `envelope.rs`
-            let envelope = NonReturningEnvelope::<A, M>::new(message);
-            self.sender
-                .send(AddressMessage::Message(Box::new(envelope)))
-                .map_err(|_| Disconnected)
-        } else {
-            Err(Disconnected)
-        }
-    }
-
-    /// Send a [`Message`](../trait.Message.html) to the actor without waiting for a response.
-    /// If the actor's mailbox is full, it will asynchronously wait. If this returns
-    /// `Err(Disconnected)`, then the actor is stopped and not accepting messages. If this returns
-    /// `Ok(())`, the will be delivered, but may not be handled in the event that the actor stops
-    /// itself (by calling [`Context::stop`](../struct.Context.html#method.stop)) before it was handled.
-    pub fn do_send_async<M>(&self, message: M) -> DoSendFuture<A>
-    where
-        M: Send + 'static,
-        A: Handler<M>,
-    {
-        if self.is_connected() {
-            let envelope = NonReturningEnvelope::<A, M>::new(message);
-            let fut = self
-                .sender
-                .clone()
-                .into_send_async(AddressMessage::Message(Box::new(envelope)));
-            DoSendFuture(DoSendFutureInner::Send(fut))
-        } else {
-            DoSendFuture(DoSendFutureInner::Disconnected)
-        }
-    }
-
-    /// Send a [`Message`](../trait.Message.html) to the actor and asynchronously wait for a response. If this
-    /// returns `Err(Disconnected)`, then the actor is stopped and not accepting messages. Like most
-    /// futures, this must be polled to actually send the message.
-    pub fn send<M>(&self, message: M) -> SendFuture<A, <A as Handler<M>>::Return>
+    /// Send a message to the actor.
+    ///
+    /// The actor must implement [`Handler<Message>`] for this to work.
+    ///
+    /// This function returns a [`Future`](SendFuture) that resolves to the [`Return`](crate::Handler::Return) value of the handler.
```

Comment:
Just a side note - I hadn't realised how powerful the new rustdoc linking was 😄 Most of the links in xtra predate its stabilisation

Reply:
They are pretty nice yeah :)